fitbenchmarking/results_processing/base_table.py

```diff
@@ -61,296 +66,356 @@ def __init__(self, results, best_results, options, group_dir,
-        into the HTML colours used in the tables. The base class
-        implementation, for example, uses the relative results and
-        ``colour_map``, ``colour_ulim`` and ``cmap_range`` within
-        :class:`~fitbenchmarking.utils.options.Options`.
-        :param results: tuple containing absolute and relative values
-        :type results: tuple
-        :return: dictionary containing HTML colours for the table
-        :rtype: dict
+        if comp_mode == "abs":
+            return result_template.format(abs_value)
+        elif comp_mode == "rel":
+            return result_template.format(rel_value)
+        # comp_mode == "both"
+        return result_template.format(abs_value, rel_value)
+
+    def get_link_str(self, result):
         """
-        cmap_name = self.options.colour_map
-        cmap = plt.get_cmap(cmap_name)
-        cmap_ulim = self.options.colour_ulim
-        cmap_range = self.options.cmap_range
-        log_ulim = np.log10(cmap_ulim)  # colour map used with log spacing
-        _, rel_value = results
-        colour = {}
-        for key, value in rel_value.items():
-            if not all(isinstance(elem, list) for elem in value):
-                hex_strs = self._vals_to_colour(value, cmap,
-                                                cmap_range, log_ulim)
-                colour[key] = hex_strs
+        Get the link as a string for the result.
+        This can be overridden if tables require different links.
+
+        :param result: The result to get the link for
+        :type result: FittingResult
+
+        :return: The link to go to when the cell is selected
+        :rtype: string
+        """
+        return os.path.relpath(path=result.support_page_link,
+                               start=self.group_dir)
+
+    def get_error_str(self, result, error_template='[{}]'):
+        """
+        Get the error string for a result based on error_template
+        This can be overridden if tables require different error formatting.
+
+        :param result: The result to get the error string for
+        :type result: FittingResult
+
+        :return: A string representation of the error
+        :rtype: str
+        """
+        error_code = result.error_flag
+        if error_code == 0:
+            return ''
+
+        return error_template.format(error_code)
+
+    def create_results_dict(self):
+        """
+        Generates a dictionary of results lists sorted into the correct order
+        with rows and columns as the key and list elements respectively.
+        This is used to create HTML and txt tables.
+        This is stored in self.sorted_results
+        """
+
+        # Might be worth breaking out into an option in future.
+        # sort_order[0] is the order of sorting for rows
+        # sort_order[1] is the order of sorting for columns
+        sort_order = (['problem'],
+                      ['software', 'minimizer', 'jacobian'])
```

Comment:
Should `sort_order` also include hessian names? Although I suppose at the moment with only one hessian method it won't make a difference

Reply:
Yes it should! I think I've hooked it up properly now